### PR TITLE
Validate 0.8 networking schema by backend

### PR DESCRIPTION
**Pull request by bbonaby**
## 📖 Description

Adds backend capability validation for schema 0.8 directional networking. Backends reject network features they do not enforce. Applies validation to one-shot and state-aware WSLC, Windows Sandbox, and IsolationSession paths.

Does not add backend enforcement.

## 🔗 References

- Stacked on #961

## 🔍 Validation

- `cargo test -p wxc_common validator::tests::`
- `cargo check -p wslc_common -p windows_sandbox_lifecycle -p isolation_session_common`
- `cargo fmt --all -- --check`

## ✅ Checklist

- [x] Signed the [Contributor License Agreement](https://cla.opensource.microsoft.com)
- [ ] Linked to an issue
- [ ] Updated documentation (if applicable)
- [ ] Updated [Copilot instructions](.github/copilot-instructions.md) (if build, architecture, or conventions changed)
- [ ] If this PR changes `Cargo.lock`, the `dependency-feed-check` check passes (see [docs/pull-requests.md](https://github.com/microsoft/mxc/blob/main/docs/pull-requests.md))

## 📋 Issue Type

- [ ] Bug fix
- [x] Feature
- [ ] Task

### CODE REVIEW

File: src/backends/windows_sandbox/lifecycle/src/state_aware.rs [hunk 1]
```diff
@@ -997,6 +998,7 @@ impl StatefulSandboxBackend for WindowsSandboxRunner {
         request: &ExecutionRequest,
         _config: Option<&()>,
     ) -> Result<(), MxcError> {
+        validate_state_aware_network_policy_support(request, NetworkPolicySupport::LEGACY)?;
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/windows_sandbox/lifecycle/src/state_aware.rs:1004]: Fixed in f4e36b6a. Windows Sandbox now rejects directional network state on every post-provision phase instead of silently ignoring it.

File: src/backends/seatbelt/common/src/seatbelt_runner.rs [hunk 2]
```diff
@@ -106,6 +106,7 @@ impl SeatbeltScriptRunner {
 
 impl SandboxBackend for SeatbeltScriptRunner {
     fn validate(&self, request: &ExecutionRequest) -> Result<(), ScriptResponse> {
+        validate_network_policy_support(request, self.network_policy_support())?;
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/seatbelt/common/src/seatbelt_runner.rs:109]: Intentionally not changed. This PR declares only capabilities the backend consumes from the new directional/runtime model today. Seatbelt legacy network.proxy behavior remains available and unchanged; runtimeConfig.networkProxy support belongs in the later Seatbelt migration PR.

File: src/backends/isolation_session/common/src/one_shot.rs [hunk 3]
```diff
@@ -56,7 +57,9 @@ impl ScriptRunner for IsolationSessionRunner {
         // backend configuration, so there is nothing backend-specific to
         // validate here — only the cross-cutting stable-surface policy.
         reject_unsupported_lifecycle(request)?;
-        validate_provision_policy(request).map_err(ScriptResponse::from)
+        validate_provision_policy(request).map_err(ScriptResponse::from)?;
+        validate_network_policy_support(request, NetworkPolicySupport::LEGACY)?;
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/isolation_session/common/src/one_shot.rs:61]: Intentionally not changed. This validation vertical does not migrate IsolationSession to the schema 0.8 directional acknowledgment. Its existing legacy canonical allow/allow behavior remains unchanged; the later backend migration will declare and validate directional default/loopback capabilities together.

File: src/backends/isolation_session/common/src/state_aware.rs [hunk 4]
```diff
@@ -181,6 +182,7 @@ impl StatefulSandboxBackend for IsolationSessionRunner {
         request: &ExecutionRequest,
         config: Option<&IsolationSessionProvisionConfig>,
     ) -> Result<(), MxcError> {
+        validate_state_aware_network_policy_support(request, NetworkPolicySupport::LEGACY)?;
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/isolation_session/common/src/state_aware.rs:185]: Intentionally not changed for the same scope reason as one-shot IsolationSession. State-aware provision keeps accepting the existing legacy acknowledgment, while schema 0.8 directional acknowledgment remains rejected until the backend-specific migration implements that contract.

File: src/core/wxc_common/src/validator.rs [hunk 5]
```diff
@@ -1,9 +1,133 @@
...
+pub fn validate_network_policy_support(
+    request: &ExecutionRequest,
+    support: NetworkPolicySupport,
+) -> Result<(), ScriptResponse> {
+    if !support.contains(NetworkPolicySupport::EGRESS_DEFAULT)
+        && request
+            .policy
+            .network_egress
+            .as_ref()
+            .is_some_and(|egress| egress.default == NetworkAction::Allow)
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/core/wxc_common/src/validator.rs]: Fixed in f4e36b6a. LEGACY validation now rejects parser-bypassing requests whose directional deny defaults contradict permissive legacy fields, covering outbound and inbound state.

File: src/backends/isolation_session/common/src/state_aware.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/isolation_session/common/src/state_aware.rs:185]: Fixed in f4e36b6a. IsolationSession post-provision validation now rejects runtime proxies and any directional network state on start, exec, stop, and deprovision.

File: src/backends/lxc/common/src/lxc_runner.rs [hunk 6]
```diff
@@ -650,6 +651,11 @@ impl LxcScriptRunner {
 }
 
 impl ScriptRunner for LxcScriptRunner {
+    fn validate_runner(&self, request: &ExecutionRequest) -> Result<(), ScriptResponse> {
+        validate_network_policy_support(request, NetworkPolicySupport::LEGACY)?;
+        Ok(())
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/lxc/common/src/lxc_runner.rs:656]: Fixed in 37d57173. Directional defaults now require their explicit EGRESS_DEFAULT / INGRESS_DEFAULT / HOST_LOOPBACK capabilities even when the value is deny. LXC remains LEGACY and therefore rejects the directional model before execution; this does not rely on legacy network_enforcement_mode, which is not part of the future directional implementation.

**Comment by bbonaby** [comment]
Addressed the latest suppressed findings in 37d57173: directional deny defaults now require explicit backend capabilities (including partial-support combinations), and every WSLC post-provision phase rejects raw directional fields while preserving the existing legacy exec proxy path.

File: src/core/wxc_common/src/validator.rs [hunk 7]
```diff
@@ -130,4 +245,101 @@ mod tests {
...
     }
+
+    #[test]
+    fn network_support_rejects_unimplemented_features() {
+        let mut request = ExecutionRequest::default();
+        request.policy.network_egress = Some(NetworkEgressPolicy {
+            default: NetworkAction::Allow,
+            ..Default::default()
+        });
+        assert!(validate_network_policy_support(&request, NetworkPolicySupport::LEGACY).is_err());
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/core/wxc_common/src/validator.rs]: Fixed in c8b07610. Each capability branch now uses an isolated request and grants only earlier capabilities needed to reach the branch under test, with assertions on the specific diagnostic.

File: src/core/wxc_common/src/validator.rs [hunk 8]
```diff
@@ -1,9 +1,178 @@
...
+    }
+}
+
+/// Reject network policy features that the selected backend cannot enforce.
+pub fn validate_network_policy_support(
+    request: &ExecutionRequest,
+    support: NetworkPolicySupport,
+) -> Result<(), ScriptResponse> {
+    let directional_posture_supplied = request.policy.network_mode_specified
+        || (request.policy.network_egress.is_some() && request.policy.network_proxy.is_enabled());
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/core/wxc_common/src/validator.rs:67]: Not changing this behavior. In this additive transition, network_mode_specified is the preserved signal that egress/ingress was explicitly supplied; the deny-valued structs are also synthesized for an omitted schema 0.8 network block. Requiring directional capabilities for that implicit shape rejected every otherwise-legacy 0.8 request and caused the LXC regression. Explicit directional fields, rules, and runtime proxy still fail closed. Backend-specific directional enforcement is intentionally deferred to the later backend migration PRs.

File: src/backends/isolation_session/common/src/policy.rs [hunk 9]
```diff
@@ -67,7 +67,14 @@ pub(super) fn validate_post_provision_policy(
...
     reject_filesystem_policy(request)?;
     reject_ui_policy(request)?;
-    if request.policy.network_specified {
+    if request.policy.network_proxy.is_enabled() {
+        return Err(IsolationSessionError::Policy(ERR_PROXY_POLICY.to_string()));
+    }
+    if request.policy.network_specified
+        || request.policy.network_mode_specified
+        || request.policy.network_egress.is_some()
+        || request.policy.network_ingress.is_some()
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/isolation_session/common/src/policy.rs:76]: False positive for the current state-aware surface. State-aware requests are still parsed as schema 0.6, so an omitted network policy leaves network_egress/network_ingress as None; the schema 0.8 one-shot normalization described here does not run. The raw Option checks catch parser-bypassing supplied directional fields, while validate_post_provision_hooks_accept_absent_network covers the inherited absent-policy path.

File: src/backends/wslc/common/src/state_aware.rs [hunk 10]
```diff
@@ -519,6 +525,24 @@ mod tests {
...
     }
 
+    #[test]
+    fn post_provision_hooks_reject_directional_network_fields() {
+        let runner = WslcStateAwareRunner::new();
+        let request = ExecutionRequest {
+            policy: ContainerPolicy {
+                network_egress: Some(NetworkEgressPolicy::default()),
+                ..Default::default()
+            },
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/wslc/common/src/state_aware.rs:535]: Fixed in 2fd57f76 without weakening the fixture. State-aware requests do not synthesize directional defaults, so a raw network_egress/network_ingress value is itself evidence of supplied post-provision policy even when a parser-bypassing request omits network_mode_specified. WSLc now rejects those raw fields in its post-provision policy helper, and the regression test passes on the Windows build targets.

File: src/backends/appcontainer/common/src/appcontainer_runner.rs [hunk 11]
```diff
@@ -1529,6 +1530,13 @@ impl AppContainerScriptRunner {
 
 impl SandboxBackend for AppContainerScriptRunner {
     fn validate(&self, request: &ExecutionRequest) -> Result<(), ScriptResponse> {
+        validate_network_policy_support(request, self.network_policy_support())?;
+        if request.policy.allowed_proxy_peer.is_some() {
```
Comment (SohamDas2021) [inline src/backends/appcontainer/common/src/appcontainer_runner.rs]: allowedProxyPeer is rejected only by the two ProcessContainer runners? The shared capability validator never checks it, so other backends that support runtime proxies can accept and silently ignore this restriction?Please model this in NetworkPolicySupport or reject it centrally.

File: src/backends/wslc/common/src/policy.rs [hunk 12]
```diff
@@ -123,10 +123,15 @@ fn reject_host_filtering(request: &ExecutionRequest) -> Result<(), MxcError> {
...
-/// block by value) is rejected too. The cooperative proxy is a separate
-/// exec-time concern handled by the callers.
+/// block by value) is rejected too. State-aware requests do not synthesize
+/// directional defaults, so raw directional fields also indicate an attempted
+/// post-provision policy change. The cooperative proxy is a separate exec-time
+/// concern handled by the callers.
 fn reject_post_provision_network_mode(request: &ExecutionRequest) -> Result<(), MxcError> {
-    if request.policy.network_mode_specified {
+    if request.policy.network_mode_specified
+        || request.policy.network_egress.is_some()
```
Comment (MGudgin) [inline src/backends/wslc/common/src/policy.rs:130]: **High (security, reliability, correctness, proportionality) - Do not treat parser-synthesized defaults as a post-provision change.**

This new presence check rests on the added comment's premise that state-aware requests do not synthesize directional defaults. The lower-stack parser does the opposite for any `0.8.0-alpha` request with no network block: it materializes `Some(default)` egress and ingress. Checked-in WSLC start/exec/stop/deprovision configs carry that version and omit network, so they now reach this branch and fail `ERR_NETWORK_IMMUTABLE` despite supplying no network policy. Windows Sandbox and IsolationSession add the same `.is_some()` checks.

The new unit tests construct `ExecutionRequest` directly, bypassing the parser and missing the integration regression.

**Fix:** Key post-provision rejection on caller-supplied intent (`network_mode_specified` or a dedicated presence marker), or preserve `None` for absent network sections on post-provision phases. Add parser-to-validator tests for empty versioned lifecycle requests on all three backends.

File: src/core/wxc_common/src/validator.rs [hunk 13]
```diff
@@ -1,9 +1,178 @@
...
+    pub const EGRESS_RULES: Self = Self(1 << 0);
+
+    /// Support for the inbound network default policy.
+    pub const INGRESS_DEFAULT: Self = Self(1 << 1);
+
+    /// Support for bidirectional host-loopback access.
+    pub const HOST_LOOPBACK: Self = Self(1 << 2);
+
+    /// Support for the runtime loopback proxy endpoint.
+    pub const RUNTIME_PROXY: Self = Self(1 << 3);
```
Comment (MGudgin) [inline src/core/wxc_common/src/validator.rs:32]: **Medium (security, cross-platform-parity) - The runtime-proxy capability does not cover all policy that must be enforced with it.**

Items 2 and 11 converge here. `allowedProxyPeer` is absent from `NetworkPolicySupport::ALL`, and only the two ProcessContainer runners remember to reject it ad hoc. Likewise, declaring `RUNTIME_PROXY` does not re-enable the existing backend allowlist, Linux credential-in-argv protection, or topology/reachability checks that remain in the lower stack's legacy-only parser branch. Today every backend is `LEGACY`, so these gaps are masked; the first backend to enable this bit can silently ignore peer identity or accept an unsafe/unreachable proxy shape.

**Fix:** Model peer-identity support explicitly and make format-independent proxy safety checks part of the capability validation path before any backend can declare `RUNTIME_PROXY`.

File: src/backends/wslc/common/src/state_aware.rs [hunk 14]
```diff
@@ -248,6 +251,7 @@ impl StatefulSandboxBackend for WslcStateAwareRunner {
         _config: Option<&()>,
     ) -> Result<(), MxcError> {
         validate_sandbox_id(sandbox_id)?;
+        validate_state_aware_network_policy_support(request, NetworkPolicySupport::LEGACY)?;
```
Comment (MGudgin) [inline src/backends/wslc/common/src/state_aware.rs:254]: **Medium (reliability) - Verify state-aware proxy policy after the real backend is known.**

This new WSLC exec validator is intended to produce the backend's authoritative unsupported-policy error, but the lower-stack parser runs first. Non-provision envelopes cannot carry `containment`, so Windows defaults them to ProcessContainer and applies ProcessContainer-only ingress/peer validation to a WSLC `runtimeConfig.networkProxy` request before this line is reached. The result is a misleading ProcessContainer parse error rather than WSLC capability validation.

This is pre-existing parser structure, but it leaves the new state-aware validation fix incomplete.

**Fix:** Carry backend context from `sandboxId`/provision into parsing, or defer backend-specific proxy validation until state-aware dispatch. Add a WSLC proxy-only exec integration test that asserts the WSLC policy-validation result.

File: src/backends/wslc/common/src/state_aware.rs [hunk 15]
```diff
@@ -228,6 +229,7 @@ impl StatefulSandboxBackend for WslcStateAwareRunner {
         request: &ExecutionRequest,
         _config: Option<&WslcProvisionPhase>,
     ) -> Result<(), MxcError> {
+        validate_state_aware_network_policy_support(request, NetworkPolicySupport::LEGACY)?;
```
Comment (MGudgin) [inline src/backends/wslc/common/src/state_aware.rs:232]: **Medium (maintainability, testability, proportionality, cross-platform-parity) - Centralize state-aware network capability validation.**

Items 4 and 14 converge on the new plumbing. WSLC repeats the same hardcoded `LEGACY` call in all five phase hooks, while Windows Sandbox and IsolationSession call the shared validator only for provision and maintain separate post-provision field lists. Those parallel definitions already disagree about synthesized defaults, which is the root of the lifecycle regression above, and tests cannot inject a capability mask through `StatefulSandboxBackend`.

**Fix:** Add a state-aware capability declaration/default and apply the shared check once in the common dispatcher for every phase; keep backend hooks only for genuinely backend-specific immutability rules.

File: src/core/wxc_common/src/sandbox_process.rs [hunk 16]
```diff
@@ -360,9 +361,20 @@ pub enum StdioMode {
...
+    ///
+    /// Return a named feature constant or compose constants with `|`. Use
+    /// [`NetworkPolicySupport::ALL`] only when every feature is enforced.
+    fn network_policy_support(&self) -> NetworkPolicySupport {
+        NetworkPolicySupport::LEGACY
+    }
+
+    /// Validates shared network support and backend-specific constraints.
+    ///
+    /// Override to add backend-specific checks. Implementations must first call
```
Comment (MGudgin) [inline src/core/wxc_common/src/sandbox_process.rs:374]: **Low (maintainability) - Make shared validation non-bypassable instead of relying on override discipline.**

The new trait contract tells every `validate` override to remember this call manually. Current implementations were updated, but a future override can silently replace the default and omit capability validation—the exact invariant this abstraction is meant to centralize.

**Fix:** Use a non-overridable validation wrapper that always runs shared network checks before delegating to a separate backend-specific hook.

**Review by MGudgin — COMMENTED** [review]
## Summary

I attribution-checked requested items 1–7 and 11–22 against PR #962's exact stacked range (`9bbd089...2fd57f7`). Five comments covering seven selected items remain attributable to #962. This is a `COMMENT` review: one integration regression is High; the rest are Medium/Low gaps in the new capability-validation architecture.

### Verification receipts

- `git merge-base 9bbd089 refs/copilot/pr-962-file` equals the PR base exactly.
- The local 19-file, +511/-22 diff matches `gh pr diff 962` after normalizing only GitHub's longer abbreviated hashes in `index` lines.
- `network_parser.rs`, `config_parser.rs`, `models.rs`, the networking guide, and the Node wire-conformance test are byte-identical base-to-head. They are not charged to #962 unless its new validator makes the pre-existing behavior newly wrong or leaves a gap in the claimed fix.
- Validator tests increase from 8 to 15; WSLC, Windows Sandbox, and IsolationSession state-aware modules each add one test. Those backend tests construct `ExecutionRequest` directly and do not cover the parser-to-validator integration that triggers the High finding.

### Findings outside the diff

None kept as standalone findings. The comments below anchor to #962's added validator/backend lines and explicitly identify any pre-existing lower-stack behavior they integrate with.

### Verified pre-existing — not attributed to this PR

- **Items 5, 6, 12, 17, 18, 20, and 22 withdrawn:** their cited parser/model/SDK code is byte-identical to #962's base; these are #961 findings or coverage requests, not #962 changes.
- **Item 13 withdrawn:** the cited BaseContainer capability-formatting helpers are outside every #962 hunk; this PR only adds shared network validation and `allowedProxyPeer` rejection near the runner's `validate` method.
- **Item 16 withdrawn:** the private FlatBuffer builders and their test seams pre-exist unchanged; #962 does not change policy serialization.
- **Item 19 withdrawn:** the IsolationSession documentation and directional parser contract are unchanged; #962 intentionally rejects unsupported directional policy and explicitly says it adds no enforcement.

### Other withdrawn findings

- **Item 7:** every backend declaring `LEGACY` is intentional under the PR's stated “does not add backend enforcement” scope; the capability lattice is future-facing, not a current functional defect by itself.
- **Item 21:** string-backed `ScriptResponse`/`MxcError::policy_validation` assertions match the repository's existing error surface; no concrete regression or untestable branch was established.

File: src/backends/appcontainer/common/src/appcontainer_runner.rs (hunk 11 above)
Comment (bbonaby) [reply src/backends/appcontainer/common/src/appcontainer_runner.rs]: Suggestion taken. Added an explicit PROXY_PEER_IDENTITY capability and centralized rejection of allowedProxyPeer when a backend does not declare support.

File: src/backends/wslc/common/src/policy.rs (hunk 12 above)
Comment (bbonaby) [reply src/backends/wslc/common/src/policy.rs:130]: Suggestion taken. Non-provision parsing now preserves absent network policy as None, and backend immutability checks use the caller-intent markers rather than normalized directional Options.

File: src/core/wxc_common/src/validator.rs (hunk 13 above)
Comment (bbonaby) [reply src/core/wxc_common/src/validator.rs:32]: Suggestion partially taken. Peer identity and runtime-proxy presence are now separate shared capabilities. Backend-specific reachability and credential/argv checks remain deferred until each backend enables directional runtime-proxy support.

File: src/backends/wslc/common/src/state_aware.rs (hunk 14 above)
Comment (bbonaby) [reply src/backends/wslc/common/src/state_aware.rs:254]: Suggestion taken. State-aware parsing now resolves the backend from sandboxId before network validation, so WSLC proxy exec reaches WSLC policy validation instead of ProcessContainer checks.

File: src/backends/wslc/common/src/state_aware.rs (hunk 15 above)
Comment (bbonaby) [reply src/backends/wslc/common/src/state_aware.rs:232]: Suggestion not taken. Centralizing all state-aware capability dispatch is a broader trait/dispatcher refactor. The focused fix removes normalized-default disagreement by preserving absence and consistently using intent markers in backend immutability hooks.

File: src/core/wxc_common/src/sandbox_process.rs (hunk 16 above)
Comment (bbonaby) [reply src/core/wxc_common/src/sandbox_process.rs:374]: Suggestion not taken. A non-overridable validation wrapper would be a broader SandboxBackend trait redesign; this PR keeps the existing contract and updates every current override.

File: src/backends/appcontainer/common/src/appcontainer_runner.rs (hunk 11 above)
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/appcontainer/common/src/appcontainer_runner.rs]: Suggestion taken. Removed the runner-specific checks; centralized network capability validation now owns this rejection.

File: src/backends/isolation_session/common/src/policy.rs [hunk 17]
```diff
@@ -67,7 +67,10 @@ pub(super) fn validate_post_provision_policy(
 ) -> Result<(), IsolationSessionError> {
     reject_filesystem_policy(request)?;
     reject_ui_policy(request)?;
-    if request.policy.network_specified {
+    if request.policy.network_proxy.is_enabled() {
+        return Err(IsolationSessionError::Policy(ERR_PROXY_POLICY.to_string()));
+    }
+    if request.policy.network_specified || request.policy.network_mode_specified {
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/backends/isolation_session/common/src/policy.rs]: Suggestion taken. Post-provision validation now checks raw directional egress and ingress values as well as presence markers, with tests covering parser-bypassing requests.

**Comment by MGudgin** [comment]
## Review re-verification after `3d47ef5`

I re-resolved the rewritten stacked range (`8fc3443...3d47ef5`) and mapped all five comments from review [#4985244179](https://github.com/microsoft/mxc/pull/962#pullrequestreview-4985244179) to the updated #961/#962 source and tests.

**Result:** 2 addressed, 1 partially addressed, 2 not addressed.

### Addressed

1. **Parser-synthesized defaults rejected as post-provision changes** (`introduced_by_change`) — updated #961 now preserves absence for versioned state-aware requests that omit network policy. #962's backend checks can therefore continue rejecting raw directional fields without rejecting ordinary lifecycle requests. Parser-to-policy tests cover omitted network and the backend tests now explicitly model raw supplied policy.
2. **State-aware proxy validated as ProcessContainer before WSLC validation** (`newly_exposed_by_change`) — updated #961 derives non-provision containment from `sandboxId`, tracks `runtimeConfig.networkProxy` presence separately, and adds a WSLC proxy-exec parser test. The #962 validator now consumes that explicit runtime-field marker.

### Partially addressed

- **Runtime-proxy capability omitted required companion policy** (`newly_exposed_by_change`) — `PROXY_PEER_IDENTITY` was added to `NetworkPolicySupport::ALL`, peer identity is rejected centrally, tests were added, and the ad hoc ProcessContainer checks were removed. The other half of the comment remains: declaring `RUNTIME_PROXY` still does not connect the format-independent backend allowlist, Linux credential-in-argv protection, or topology/reachability checks that remain below the legacy-format branch.

### Not addressed

- **State-aware capability validation remains duplicated and asymmetric** (`introduced_by_change`) — WSLC still hardcodes the shared `LEGACY` call in all five hooks, while Windows Sandbox and IsolationSession use the shared validator only at provision and retain separate post-provision field lists. No injectable state-aware capability declaration or dispatcher-level validation was added.
- **Shared `SandboxBackend` validation remains bypassable by overrides** (`introduced_by_change`) — the trait still requires each `validate` override to remember to call the shared validator manually; there is no non-overridable wrapper/backend-specific hook split.

### Verification

`cargo test -p wxc_common -p wslc_common -p windows_sandbox_lifecycle -p isolation_session_common` at the updated head:

- **1,460 passed**
- **0 failed**
- **1 ignored**

The original final attribution values remain unchanged; the statuses above describe resolution at the updated head.

**Review by MGudgin — APPROVED** [review]
